File: YJPageView/YJPageView/YJImagePageViewController.m
//
//  YJImagePageViewController.m
//  YJPageView
//
//  CSDN:http://blog.csdn.net/y550918116j
//  GitHub:https://github.com/937447974
//
//  Created by 阳君 on 16/5/3.
//  Copyright © 2016年 YJ. All rights reserved.
//

#import "YJImagePageViewController.h"

@interface YJImagePageViewController ()

@property (weak, nonatomic) IBOutlet UIImageView *imageView;

@end

@implementation YJImagePageViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
}

- (void)reloadPageWithPageViewObject:(YJPageViewObject *)pageViewObject pageView:(YJPageView *)pageView {
    
    [super reloadPageWithPageViewObject:pageViewObject pageView:pageView];
    NSLog(@"%ld", (long)pageViewObject.pageIndex);
    self.imageView.image = [UIImage imageNamed:@"LaunchImage"];
    
}

#pragma mark 点击事件
- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    
    [super touchesBegan:touches withEvent:event];
    self.pageView.pageViewDidSelect(self);
    
}

@end

File: YJPageViewSwift/YJPageViewSwift/YJImagePageViewController.swift
//
//  YJImagePageViewController.swift
//  YJPageViewSwift
//
//  CSDN:http://blog.csdn.net/y550918116j
//  GitHub:https://github.com/937447974
//
//  Created by 阳君 on 16/5/4.
//  Copyright © 2016年 YJ. All rights reserved.
//

import UIKit

class YJImagePageViewController: YJPageViewController {

    @IBOutlet weak var imageView: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override class func pageViewObject() -> YJPageViewObject {
        let pvo = super.pageViewObject()
        pvo.createPageView = YJPageViewCreate.Xib
        return pvo
        
    }
    
    override func reloadPageWithPageViewObject(pageViewObject: YJPageViewObject, pageView: YJPageView) {
        super.reloadPageWithPageViewObject(pageViewObject, pageView: pageView)
        self.imageView.image = UIImage(named: "LaunchImage")
    }

}
